src/PostgREST/QueryBuilder.hs

```diff
@@ -153,23 +153,18 @@ requestToCallProcQuery qi pgArgs returnsScalar preferParams returnings =
     sourceBody :: SqlFragment
     sourceBody
       | paramsAsMultipleObjects =
-          if returnsScalar
-            then "SELECT " <> callIt <> " AS pgrst_scalar FROM pgrst_args"
-            else BS.unwords [ "SELECT pgrst_lat_args.*"
-                         , "FROM pgrst_args,"
-                         , "LATERAL ( SELECT " <> returned_columns <> " FROM " <> callIt <> " ) pgrst_lat_args" ]
-      | otherwise =
-          if returnsScalar
-            then "SELECT " <> callIt <> " AS pgrst_scalar"
```

Comment:
Note: Having the function call in the select expression list, does also potentially harm performance, because inlining the function is much less likely, given the huge list of conditions:
https://wiki.postgresql.org/wiki/Inlining_of_SQL_functions

Especially this part:
> A [...] function call will be inlined if all of the following conditions are met:
> [...]
> the body contains no aggregate or window function calls, no subqueries, no CTEs, no FROM clause or references to any table or table-like object, none of GROUP BY, HAVING, ORDER BY, DISTINCT, LIMIT, OFFSET, UNION, INTERSECT, EXCEPT

Having the function call in the `FROM` part, will allow much more. So this PR potentially also improves performance, also currently only in theory.

That's, because currently, inlining of non-scalar functions is still prevented by the sub-select in the argument list (`x := (SELECT x FROM pgrst_args LIMIT 1)`, see https://github.com/PostgREST/postgrest/issues/1652#issuecomment-729975165. Once we can fix that and make the change in this PR here, that should lead to better performance for many RPCs.

Reply:
Scratch that. Since this is only about scalar functions, those won't be inlined in the `FROM` part at all. So if anything, this change might potentially harm inlining.

I guess I have to re-think the approach in this PR.